test/genrconfig_test.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# mockexec_test.sh exercises the mock executable and verifies that it behaves as expected.
+# It takes no inputs from commandline, prints issues to stdout and returns a non-zero
+# exit status when expectations are not met.
+
+EXIT_STATUS=0
+
+# correct usage testcase 1 (with output path)
+echo "INPUT1" > tmp/test-input1
+echo "INPUT2" > tmp/test-input2
+cmd/genrconfig/genrconfig examples/tar tar "strarg:-cf" "outpath:tmp/output" "infile:tmp/test-input1" "infile:tmp/test-input2" > /dev/null 2>&1
+if [ $? -ne 0 ]
```

Comment:
One more check you should do is to compare the content of the generated proto to the expected proto. You could do that by just adding the expected proto config files (under test/data for example) and then compare that the files have the same content.

Reply:
you mean in the “if” clause?

Reply:
yes, in that case. But it could probably also be another if clause, but you want to first make sure the tool did produce something, and once it did, you want to check that the content matches what you expected.

Reply:
okay